src/Maestro/Maestro.DataProviders/ConfigurationIngestion/ConfigurationDataHelper.cs

```diff
@@ -0,0 +1,178 @@
+
+        IEnumerable<T> removals = [.. dbEntitiesById.Values
+            .Where(e => !externalEntitiesById.ContainsKey(e.UniqueId))];
+
+        IEnumerable<T> updates = [.. externalEntitiesById.Values
+            .Where(e => dbEntitiesById.ContainsKey(e.UniqueId))];
+
+        return new EntityChanges<T>(creations, updates, removals);
+    }
+
+    internal static Subscription ConvertIngestedSubscriptionToDao(
+        IngestedSubscription subscription,
+        Namespace namespaceEntity,
+        Dictionary<string, Channel> existingChannelsByName)
+    {
+        existingChannelsByName.TryGetValue(subscription.Values.Channel, out Channel? existingChannel);
+
+        if (existingChannel is null)
+        {
+            //todo find the right exception type
+            throw new InvalidOperationException(
+                $"Channel '{subscription.Values.Channel}' not found for subscription creation.");
+        }
+
+        return new Subscription
+        {
+            Id = subscription.Values.Id,
+            ChannelId = existingChannel.Id,
+            Channel = existingChannel,
+            SourceRepository = subscription.Values.SourceRepository,
+            TargetRepository = subscription.Values.TargetRepository,
+            TargetBranch = subscription.Values.TargetBranch,
+            PolicyObject = new SubscriptionPolicy
+            {
+                UpdateFrequency = (UpdateFrequency)(int)subscription.Values.UpdateFrequency,
+                Batchable = subscription.Values.Batchable,
+                MergePolicies = [.. subscription.Values.MergePolicies.Select(ConvertMergePolicyYamlToDao)],
+            },
+            Enabled = subscription.Values.Enabled,
+            SourceEnabled = subscription.Values.SourceEnabled,
+            SourceDirectory = subscription.Values.SourceDirectory,
+            TargetDirectory = subscription.Values.TargetDirectory,
+            PullRequestFailureNotificationTags = subscription.Values.FailureNotificationTags,
+            ExcludedAssets = subscription.Values.ExcludedAssets == null ? [] : [.. subscription.Values.ExcludedAssets.Select(asset => new AssetFilter() { Filter = asset })],
+            Namespace = namespaceEntity,
+        };
+    }
+
+    internal static Channel ConvertIngestedChannelToDao(
+        IngestedChannel channel,
+        Namespace namespaceEntity)
+        =>  new()
+        {
+            Name = channel.Values.Name,
+            Classification = channel.Values.Classification,
+            Namespace = namespaceEntity,
+        };
+
+    internal static DefaultChannel ConvertIngestedDefaultChannelToDao(
+        IngestedDefaultChannel defaultChannel,
+        Namespace namespaceEntity,
+        Dictionary<string, Channel> existingChannelsByName,
+        Dictionary<(string, string, string), IngestedDefaultChannel>? existingDefaultChannels)
+    {
+        existingChannelsByName.TryGetValue(defaultChannel.Values.Channel, out Channel? existingChannel);
```

Comment:
you can do `if (existinChannelsByName.TryGetValue(....)`
and remove the if below